service/service.go

```diff
@@ -70,7 +70,36 @@ func NewService(options Options, registry *plugin.Registry) *Service {
 }
 
 func (s *Service) Start() error {
-	s.initLogger()
+	// if .LogFormatter is set, it'll be used in log.SetFormatter() and .Log will be ignored.
+	if s.options.LogFormatter != nil {
+		log.SetFormatter(s.options.LogFormatter)
+	} else {
+		switch s.options.Log {
+			case "console": {
+				log.SetFormatter(&log.TextFormatter{})
+			}
+			case "json": {
+				log.SetFormatter(&log.JSONFormatter{})
+			}
+			case "syslog": {
+				hook, err := logrus_syslog.NewSyslogHook("", "", syslog.LOG_INFO, "")
```

Comment:
we should never fail if syslog is not available, check out the code in `initLogger` for detail.